### PR TITLE
Fix help message for GCS

### PR DESCRIPTION
**Pull request by krishnasrinivas**
## Description

`gcloud init` or `gcloud beta auth application-default login` way of providing credentials in not suitable for applications like minio.

When users want to deploy minio gateway on VMs or using docker containers we can't expect them to do "gloud init" the deployment would have to be automated/scriptable.

https://cloud.google.com/storage/docs/authentication the section Other Environments points to https://developers.google.com/identity/protocols/application-default-credentials#howtheywork which has instructions on how to use GOOGLE_APPLICATION_CREDENTIALS This is most suitable for applications like minio.

I have edited the help message accordingly. We can use the same way for our testing as well, i.e the admin will have to generate and give us credentials.json for testing.

and:

```
export GOOGLE_APPLICATION_CREDENTIALS=/home/krishna/Downloads/miniotest-d4530e7be663.json
minio gateway gcs PROJECTID
```

## Motivation and Context

https://github.com/minio/minio/issues/4391

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

**Review by balamurugana — COMMENTED** [review]
What will happen if `gcloud init --console-only` is used instead of `GOOGLE_APPLICATION_CREDENTIALS`?

File: cmd/gateway-main.go [hunk 1]
```diff
@@ -63,7 +63,8 @@ EXAMPLES:
       $ {{.HelpName}} s3 https://play.minio.io:9000
 
   4. Start minio gateway server for GCS backend.
-      $ gcloud init --console-only # this will configure your google account
+      $ export GOOGLE_APPLICATION_CREDENTIALS=/path/to/credentials.json
```
Comment (krisis) [inline cmd/gateway-main.go]: What a user needs to do for Minio GCS gateway to successfully authenticate with the GCS account depends on the environment it is being run. This message applies only when the gateway is run _outside_ GCP. Is it possible for the gateway to know if it's running in a GCP instance? If not, we could modify the message to,

> (When run outside Google Cloud Platform)
$ export GOOGLE_APPLICATION_CREDENTIALS=/path/to/credentials.json

**Review by krisis — CHANGES_REQUESTED** [review]

**Comment by harshavardhana** [comment]
@krishnasrinivas can you rebase this ?

File: cmd/gateway-main.go (hunk 1 above)
Comment (krishnasrinivas) [reply cmd/gateway-main.go]: Not true @krisis this step works everywhere

**Comment by krishnasrinivas** [comment]
This PR is temporary fix till we fix https://github.com/minio/minio/issues/4457

**Comment by krishnasrinivas** [comment]
@harshavardhana done

**Comment by harshavardhana** [comment]
Is this still valid @krishnasrinivas ?

**Pull request closed without merging**